semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/nodemethods/MethodMethods.scala

```diff
@@ -60,17 +51,16 @@ class MethodMethods(val method: Method) extends AnyVal with NodeExtension with H
 
   /** The type declaration associated with this method, e.g., the class it is defined in.
     */
-  def definingTypeDecl: Traversal[TypeDecl] =
-    Traversal.fromSingle(method).definingTypeDecl
+  def definingTypeDecl: Option[TypeDecl] =
+    Traversal.fromSingle(method).definingTypeDecl.headOption
```

Comment:
The steps on the traversals should be defined in terms of the steps on a single element and not the other way around. I know this was already not the case before your change but since you are already touching it.

Reply:
I was considering that, but some steps (like this one) are defined with a repeat/until traversal, and it's probably best to keep it that way, since those are quite expressive/readable...